src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java

```diff
@@ -0,0 +1,87 @@
+package tv.rewinside.nbtstorage.nbt;
+
+import tv.rewinside.nbtstorage.exceptions.NBTLoadException;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class NBTTagLongArray extends NBTBase {
+
+	private long[] data;
+
+	NBTTagLongArray() {
+	}
+
+	public NBTTagLongArray(long[] along) {
+		this.data = along;
+	}
+
+	@Override
+	void write(DataOutput dataoutput) throws IOException {
+		dataoutput.writeInt(this.data.length);
+
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
+	}
+
+	@Override
+	void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
```

Comment:
some other name, what exactly is "i"?
datainput -> dataInput
nbtreadlimiter -> nbtReadLimiter